runconfig/config_test.go

```diff
@@ -30,16 +49,236 @@ func TestDecodeContainerConfig(t *testing.T) {
+	}
+
+	// Root to non-c: drive letter (Windows specific)
+	if runtime.GOOS == "windows" {
+		tryit = volSetup([]string{}, []string{os.Getenv("SystemDrive") + `\:d:`})
+		if config, hostConfig, err = callDCC(tryit); err != nil {
+			t.Fatal(err)
+		}
+		if hostConfig.Binds == nil || len(hostConfig.Binds) > 1 || hostConfig.Binds[0] != tryit[0] || len(config.Volumes) != 0 {
+			t.Fatalf("Error parsing %s. Should have a single bind mount and no volumes", tryit[0])
+		}
+	}
+
+	//
+	// Error cases
+	//
+
+	// Empty spec
+	tryit = []string{}
+	if config, hostConfig, err = callDCC(tryit); err != nil {
+		if hostConfig.Binds != nil || len(config.Volumes) != 0 {
+			t.Fatalf("Empty volume spec should have no binds or volumes. Received %s %q", hostConfig.Binds, config.Volumes)
+		}
+	}
+
+	// Root to root
+	tryit = volSetup([]string{`/:/`}, []string{os.Getenv("SystemDrive") + `\:c:\`})
+	if config, hostConfig, err = callDCC(tryit); err == nil {
+		t.Fatalf("%s should have failed", tryit[0])
+	}
+
+	// No destination path
+	tryit = volSetup([]string{`/tmp:`}, []string{os.Getenv("TEMP") + `\:`})
+	if config, hostConfig, err = callDCC(tryit); err == nil {
+		t.Fatalf("%s should have failed", tryit[0])
+	}
+
+	// No destination path or mode
+	tryit = volSetup([]string{`/tmp::`}, []string{os.Getenv("TEMP") + `\::`})
+	if config, hostConfig, err = callDCC(tryit); err == nil {
+		t.Fatalf("%s should have failed", tryit[0])
+	}
+
+	// A whole lot of nothing
+	tryit = []string{`:`}
+	if config, hostConfig, err = callDCC(tryit); err == nil {
+		t.Fatalf("%s should have failed", tryit[0])
+	}
+
+	// A whole lot of nothing with no mode
+	tryit = []string{`::`}
+	if config, hostConfig, err = callDCC(tryit); err == nil {
+		t.Fatalf("%s should have failed", tryit[0])
+	}
+
+	// Too much including an invalid mode
+	wTmp := os.Getenv("TEMP")
+	tryit = volSetup([]string{`/tmp:/tmp:/tmp:/tmp:`}, []string{wTmp + ":" + wTmp + ":" + wTmp + ":" + wTmp})
+	if config, hostConfig, err = callDCC(tryit); err == nil {
+		t.Fatalf("%s should have failed", tryit[0])
+	}
+
+	// Windows specific error tests
+	if runtime.GOOS == "windows" {
+		// Volume which does not include a drive letter
+		tryit = []string{"\tmp"}
+		if config, hostConfig, err = callDCC(tryit); err == nil {
+			t.Fatalf("%s should have failed", tryit[0])
+		}
+
+		// Root to C-Drive
+		if runtime.GOOS == "windows" {
+			tryit = volSetup([]string{}, []string{os.Getenv("SystemDrive") + `\:c:`})
+			if config, hostConfig, err = callDCC(tryit); err == nil {
+				t.Fatalf("%s should have failed", tryit[0])
+			}
+		}
+
+		// Container path that does not include a drive letter
+		tryit = []string{"c:\tmp:\tmp"}
+		if config, hostConfig, err = callDCC(tryit); err == nil {
+			t.Fatalf("%s should have failed", tryit[0])
+		}
+	}
+}
+
+func volSetup(u []string, w []string) []string {
+	if runtime.GOOS == "windows" {
+		return w
+	}
+	return u
+}
+
+func callDCC(volumes []string) (*Config, *HostConfig, error) {
```

Comment:
DCC was DecodeContainerConfig. You want me to change it to callDecodeContainerConfig?